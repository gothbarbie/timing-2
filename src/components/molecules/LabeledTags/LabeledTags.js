import React from 'react'
import styled from 'styled-components'

import Label from '../../atoms/Label'
import Tag from '../../atoms/Tag'

const LabeledTagsWrapper = styled.div`
  display: flex;
  flex-direction: column;
  align-items: flex-start;
`
const LabelStyled = styled(Label)`
  margin-bottom: 1rem;
`
const LabeledTags = ({ label, tags, onClick }) => (
  <LabeledTagsWrapper>
    <LabelStyled>{label}</LabelStyled>
    {tags.map(tag => (
      <Tag
        blue={tag.blue}
        key={tag.id}
        name={tag.name}
        onClick={() => onClick(tag.id)}
      />
    ))}
  </LabeledTagsWrapper>
)

export default LabeledTags
