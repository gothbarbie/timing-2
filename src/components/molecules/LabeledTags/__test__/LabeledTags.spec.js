import React from 'react'
import { shallow } from 'enzyme'

import LabeledTags from '../LabeledTags'

describe('components/molecules/LabeledTags', () => {
  it('renders', () => {
    const tags = [
      {
        blue: true,
        name: 'tag 1',
        id: '1',
      },
    ]

    const props = {
      label: 'Test Tag',
      onClick: jest.fn(),
      tags,
    }

    const component = shallow(<LabeledTags {...props} />)

    expect(component).toMatchSnapshot()
  })
})
