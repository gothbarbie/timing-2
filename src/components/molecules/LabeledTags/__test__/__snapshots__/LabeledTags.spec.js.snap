// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/molecules/LabeledTags renders 1`] = `
<LabeledTagsWrapper>
  <LabelStyled>
    Test Tag
  </LabelStyled>
  <Tag
    blue={true}
    key="1"
    name="tag 1"
    onClick={[Function]}
  />
</LabeledTagsWrapper>
`;
